doc/manual/packages.rst

```diff
@@ -621,11 +629,22 @@ not domain experts*.
 
   * ``MATLAB.jl`` provides an interface to call the MATLAB engine from within Julia.
 
+6. Creative names are cool, but confusion isn't so much.
+
+  * Many modern programming languages such as Ruby and Python have popular packages
+    that have entertaining names. Python's ``pickle`` package isn't about pickles, it's
+    for serializing (or pickling) object structures. Ruby's ``Sinatra`` package is named
+    after the popular American singer and is a popular web framework that competes with Ruby's
+    ``Rails`` web framework (another creative name).
+
+  * If being a creative package namer isn't your style, it's okay to not force it.
```

Comment:
I'm not opposed to creative names when no obvious descriptive name comes to mind, but I'd simply recommend using explicit names where possible, that's all. For example, I'm glad `DataFrames.jl` is called that way rather than `Pandas.jl`.

(I don't think the marketing plays a big role when there are several thousands packages like in R's CRAN. At some point there is too much information for you to care...)